tests/test_events_api.py

```diff
@@ -0,0 +1,137 @@
+        assert self.event_api._bus_name == "million-dollar-lambda-event-bus"
+        assert self.event_api._resources == []
+
+    def test_set_source(self) -> None:
+        # pylint: disable=protected-access
+        self.event_api.set_source("XXX")
+        assert self.event_api._source == "XXX"
+
+    def test_set_resource(self) -> None:
+        # pylint: disable=protected-access
+        self.event_api.set_resources(["x", "Y"])
+        assert self.event_api._resources == ["x", "Y"]
+
+    def test_register(self) -> None:
+        # pylint: disable=protected-access
+        event = MyTestEvent({"x": 1})
+        self.event_api.register(event)
+
+        assert self.event_api._pending_events == [event]
+
+    def test_get_all_registered_events(self) -> None:
+        event_1 = MyTestEvent({"x": 1})
+        event_2 = MyTestEvent({"x": 2})
+        self.event_api.register(event_1)
+        self.event_api.register(event_2)
+        assert self.event_api.get_all_pending_events() == [event_1, event_2]
+
+    @patch.object(Boto3Client, "eventbridge")
+    @patch.object(EventAPI, "_mark_sent")
+    def test_send(self, mock__mark_sent: MagicMock, mock_send: MagicMock) -> None:
+        event = MyTestEvent({"x": 1})
+        self.event_api.register(event)
+
+        self.event_api.send()
+
+        mock_send.put_events.assert_called_once_with(
+            Entries=[
+                {
+                    "Detail": '{"x": 1}',
+                    "DetailType": "MY_TEST_EVENT",
+                    "EventBusName": "million-dollar-lambda-event-bus",
+                    "Resources": [],
+                    "Source": "million-dollar-lambda",
+                }
+            ]
+        )
+        mock__mark_sent.assert_called_once()
+
+    def test_clear(self):
+        # pylint: disable=protected-access
+        event = MyTestEvent({"x": 1})
+        self.event_api.register(event)
+
+        self.event_api._mark_sent()
+
+        assert self.event_api._pending_events == []
+        assert self.event_api._sent_events == [event]
+
+    def test_get_all_sent_events(self):
+        event = MyTestEvent({"x": 1})
+        self.event_api.register(event)
+
+        self.event_api._mark_sent()  # pylint: disable=protected-access
+
+        assert self.event_api.get_all_sent_events() == [event]
+
+    @patch.object(Boto3Client, "eventbridge")
+    def test_send_no_events(self, mock_send: MagicMock) -> None:
+        self.event_api.send()
+
+        mock_send.assert_not_called()
+
+    def test_create_eb_entry(self) -> None:
+        new_event = MyTestEvent({"x": 1})
+
+        resp = self.event_api._create_eb_entry(new_event)  # pylint: disable=protected-access
+
+        assert resp == {
+            "Detail": '{"x": 1}',
+            "DetailType": "MY_TEST_EVENT",
+            "EventBusName": "million-dollar-lambda-event-bus",
+            "Resources": [],
+            "Source": "million-dollar-lambda",
+        }
+
+
+class TestEventApiSingleton:
```

Comment:
This should be an integral part of the above class as you still test `EventAPI`.